### PR TITLE
check an invariant of class IndexSet

### PR DESCRIPTION
**Pull request by fernandohv3279**

### CODE REVIEW

**Comment by bangerth** [comment]
We practiced writing tests in my research group meeting today. This is the outcome. I would appreciate if someone else took a look.

**Review by drwells — APPROVED** [review]
Nice!

File: tests/base/index_set_empty.output [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+DEAL::OK
+DEAL::OK
+DEAL::OK
+DEAL::OK
+DEAL::OK
```
Comment (drwells) [inline tests/base/index_set_empty.output:6]:
```suggestion
```

**Review by drwells — CHANGES_REQUESTED** [review]

**Review by drwells — APPROVED** [review]

**Pull request merged**